src/main/java/org/apache/commons/io/output/DeferredFileOutputStream.java

```diff
@@ -213,6 +216,13 @@ protected OutputStream getStream() throws IOException
      *
      * @throws IOException if an error occurs.
      */
+    // memoryOutputStream is assigned null value after this methods is invoked. After that 
+    // memoryOutputStream not used further and remaining data is written on disk based storage.
+    // all the warnings for dereference of nullable for memoryOutputStream are false positive
+    // since memoryOutputStream when null is never used.
+    // If prefix is null, this class throws IllegalArgumentException and if prefix is non-null
```

Comment:
I don't understand this comment.  A class doesn't throw an exception, a method does.  Some constructors force `prefix` to be non-null, but not all of them do, and `prefix` is annotated as `@Nullable`.  This comment suggests that `prefix` is non-null, but that isn't currently true.  If you want it to be true, you need to change annotations and possibly constructors.

Reply:
I have made correction in annotation for parameter in constructor, My assumption about prefix to be null in all constructors was wrong. The overloading of constructors is such that only one from `prefix` or `outputFile` can have null value.

Reply:
Javadoc for this `thresholdReached()` method 
```
/**
     * Switches the underlying output stream from a memory based stream to one
     * that is backed by disk. This is the point at which we realise that too
     * much data is being written to keep in memory, so we elect to switch to
     * disk-based storage.
     *
     * @throws IOException if an error occurs.
     */
```
Is it right to say that this stream is closed after invoking `thresholdReached()` . `memoryOutputStream` is assigned null value in this method, but javadoc suggests that discontinuity in using stream to write data, so it will be correct to suppress warnings for `memoryOutputStream`.

Reply:
The Javadoc doesn't say it is closed, and the code seems to change its representation but keep it open.

Reply:
In this case, the use/property of `memoryOutputStream` after invocation of this method is unclear. I researched for test-case but was not able to produce buggy code.I will reset 1 warning suppressed for `memoryOutputStream` as it possibly can be bug.